lib/Parameters/BGV/Params.h

```diff
@@ -0,0 +1,86 @@
+#ifndef LIB_PARAMETERS_BGV_PARAMS_H_
+#define LIB_PARAMETERS_BGV_PARAMS_H_
+
+#include <cmath>
+#include <vector>
+
+#include "llvm/include/llvm/Support/raw_ostream.h"  // from @llvm-project
+
+namespace mlir {
+namespace heir {
+namespace bgv {
+
+class SchemeParam {
+ public:
+  SchemeParam(int ringDim, int64_t plaintextModulus, int level,
+              const std::vector<int> &logqi, int dnum,
+              const std::vector<int> &logpi)
+      : ringDim(ringDim),
+        plaintextModulus(plaintextModulus),
+        level(level),
+        logqi(logqi),
+        dnum(dnum),
+        logpi(logpi) {}
+
+ private:
+  // the N in Z[X]/(X^N+1)
+  int ringDim;
+
+  // the plaintext modulus for BGV
+  int64_t plaintextModulus;
+
+  // RNS level, from 0 to L
+  int level;
+
+  // logarithm of the modulus of each level
+  // logqi.size() == level + 1
+  std::vector<int> logqi;
+
+  // number of digits
+  int dnum;
+  // logarithm of the special modulus
+  // used during key switching
+  std::vector<int> logpi;
+
+  // the standard deviation of the error distribution
+  double std0 = 3.2;
+
+ public:
+  int getRingDim() const { return ringDim; }
+  int64_t getPlaintextModulus() const { return plaintextModulus; }
+  int getLevel() const { return level; }
+  const std::vector<int> &getLogqi() const { return logqi; }
+  int getDnum() const { return dnum; }
+  const std::vector<int> &getLogpi() const { return logpi; }
+  double getStd0() const { return std0; }
+
+  void print(llvm::raw_ostream &os) const;
+
+  static SchemeParam getConservativeSchemeParam(int level,
+                                                int64_t plaintextModulus);
+};
+
+class LocalParam {
```

Comment:
Please add a comment above the class explaining what "local" means. I think it means: the set of parameters specific to an SSA value at a particular point in the program execution.